### PR TITLE
Add snippet updater

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: scripts/python/update_snippet/python_snippet_updater.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            not_up_to_date = True
+            content = content.replace(body, target_code)
+    with open(file_obj, 'w') as f:
+        f.write(content)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("path", help="The path to your service folder")
+    args = parser.parse_args()
```
Comment (scbedd) [inline scripts/python/update_snippet/python_snippet_updater.py:82]: nit. consistency. If you're going to use argparse, you may as well use it all the way 👍 

```python
parser.add_argument(
        "path",
        nargs="?",
        help=(
            "The targeted path for update."
        ),
    )
```

File: scripts/python/update_snippet/test_python_snippet_updater.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    handler = logging.StreamHandler(stream=sys.stdout)
+    logger.addHandler(handler)
+    folder = os.path.dirname(os.path.abspath(__file__))
+    file = os.path.join(folder, "example_async.py")
+    get_snippet(file)
+    file_1 = os.path.join(folder, "README.md")
+    update_snippet(file_1)
+
+def test_missing_snippet():
+    logger = logging.getLogger("azure")
```
Comment (scbedd) [inline scripts/python/update_snippet/test_python_snippet_updater.py:29]: nit: do these lines do anything for you?

```
    logger = logging.getLogger("azure")
    handler = logging.StreamHandler(stream=sys.stdout)
    logger.addHandler(handler)
```

File: scripts/python/update_snippet/python_snippet_updater.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    with open(file_obj, 'r') as f:
+        content = f.read()
+    pattern = "# \\[START(?P<name>[A-Z a-z0-9_]+)\\](?P<body>[\\s\\S]+?)# \\[END[A-Z a-z0-9_]+\\]"
+    matches = re.findall(pattern, content)
+    for match in matches:
+        s = match
+        name = s[0].strip()
+        snippet = s[1]
+        # Remove extra spaces
+        spaces = ""
```
Comment (scbedd) [inline scripts/python/update_snippet/python_snippet_updater.py:33]: I'd really appreciate a little code comment or something that maps out what is actually happening here. I am not convinced of this code as-is just from reading it.

A snippet is the complete code from the actual python file. So ripped directly from the python file it'll look like.

```
        # [START trio]
        from azure.core.pipeline.transport import TrioRequestsTransport

        async with AsyncPipeline(TrioRequestsTransport(), policies=policies) as pipeline:
            return await pipeline.run(request)
        # [END trio]
```

And we need to _remove_ spaces, as we will be emplacing the snippet into the markdown file.

First question, why do we start with the second character of the snippet? Shouldn't we start with the first?

```
        if snippet[-1] == "\n":
            snippet = snippet[:-1]
```

I seriously do not get it.

**Review by scbedd — CHANGES_REQUESTED** [review]
I'm too dumb to track what the code is actually doing with the indexes, needs some explanation.

**Comment by xiangyan99** [comment]
Close the PR and use https://github.com/Azure/azure-sdk-for-python/pull/26899 to track the code change.

**Pull request closed without merging**